app/src/main/java/com/arctouch/codechallenge/view/home/HomeActivity.java

```diff
@@ -0,0 +1,108 @@
+package com.arctouch.codechallenge.view.home;
+
+import androidx.lifecycle.ViewModelProviders;
+import android.content.Intent;
+import android.os.Bundle;
+import androidx.annotation.Nullable;
+import androidx.core.view.MenuItemCompat;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.appcompat.widget.SearchView;
+
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.data.model.Movie;
+import com.arctouch.codechallenge.view.home.adpters.HomeAdapter;
+import com.arctouch.codechallenge.view.home.viewmodel.HomeActivityViewModel;
+
+public class HomeActivity extends AppCompatActivity implements View.OnClickListener,
+                                                                SearchView.OnQueryTextListener{
+
+    private RecyclerView recyclerView;
+    private ProgressBar progressBar;
+    private HomeActivityViewModel viewModel;
+    private HomeAdapter homeAdapter;
+
+        @Override
+        protected void onCreate(@Nullable Bundle savedInstanceState) {
+            super.onCreate(savedInstanceState);
+            setContentView(R.layout.home_activity);
+
+            this.recyclerView = findViewById(R.id.recyclerView);
+            this.progressBar = findViewById(R.id.progressBar);
+
+            viewModel = ViewModelProviders.of(this).get(HomeActivityViewModel.class);
+            viewModel.init();
+
+            viewModel.getMoviePagedList().observe(this, movies -> {
+                this.progressBar.setVisibility(View.GONE);
+                homeAdapter.submitList(movies);
+            });
+
+        viewModel.getErrorHandlerMutable().observe(this, throwableMutableLiveData -> {
+            throwableMutableLiveData.observe(this, throwable -> {
+                Toast.makeText(this, "An Error Has Ocurred", Toast.LENGTH_LONG).show();
+            });
+        });
+
+        homeAdapter =  new HomeAdapter(this);
+        recyclerView.setAdapter(homeAdapter);
+    }
+
+    @Override
+    public void onClick(View v) {
```

Comment:
Why did you add a `OnClickListener` to this `Activity` instead of adding one to each item in your `RecyclerView`.

Reply:
In my opinion, it is just a matter of preference because if I had this onClickLitener inside of my adapter, I would have done an interface with just the callback method to call showMovieDetail that redirects the user to the Detail Screen